File: go-sdk/pkg/models/shared/skillcardsmetadata.go
package shared

// SkillCardsMetadata
// The metadata assigned to a using for Box skills.
type SkillCardsMetadata struct {
	DollarCanEdit     *bool         `json:"$canEdit,omitempty"`
	DollarID          *string       `json:"$id,omitempty"`
	DollarParent      *string       `json:"$parent,omitempty"`
	DollarScope       *string       `json:"$scope,omitempty"`
	DollarTemplate    *string       `json:"$template,omitempty"`
	DollarType        *string       `json:"$type,omitempty"`
	DollarTypeVersion *int64        `json:"$typeVersion,omitempty"`
	DollarVersion     *int64        `json:"$version,omitempty"`
	Cards             []interface{} `json:"cards,omitempty"`
}

File: ts-sdk/src/sdk/models/shared/skillcardsmetadata.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { SkillCard } from "./skillcard";
import { KeywordSkillCard } from "./keywordskillcard";
import { TimelineSkillCard } from "./timelineskillcard";
import { TranscriptSkillCard } from "./transcriptskillcard";
import { StatusSkillCard } from "./statusskillcard";



// SkillCardsMetadata
/** 
 * The metadata assigned to a using for Box skills.
**/
export class SkillCardsMetadata extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=$canEdit" })
  dollarCanEdit?: boolean;

  @SpeakeasyMetadata({ data: "json, name=$id" })
  dollarId?: string;

  @SpeakeasyMetadata({ data: "json, name=$parent" })
  dollarParent?: string;

  @SpeakeasyMetadata({ data: "json, name=$scope" })
  dollarScope?: string;

  @SpeakeasyMetadata({ data: "json, name=$template" })
  dollarTemplate?: string;

  @SpeakeasyMetadata({ data: "json, name=$type" })
  dollarType?: string;

  @SpeakeasyMetadata({ data: "json, name=$typeVersion" })
  dollarTypeVersion?: number;

  @SpeakeasyMetadata({ data: "json, name=$version" })
  dollarVersion?: number;

  @SpeakeasyMetadata({ data: "json, name=cards" })
  cards?: any[];
}
